pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -465,6 +465,11 @@ type RolloutStatus struct {
 	PauseConditions []PauseCondition `json:"pauseConditions,omitempty"`
 	//ControllerPause indicates the controller has paused the rollout
 	ControllerPause bool `json:"controllerPause,omitempty"`
+	// ReconciledAbort indicates the controller reconciled an aborted rollout. The controller uses this to understand if
+	// the controller needs to do some specific work when a Rollout is aborted. For example, the reconcileAbort is used
+	// to indicate if the Rollout should enter an aborted state when the latest AnalysisRun is a failure, or the controller
+	// has already put the Rollout into an aborted and should create a new AnalysisRun.
+	ReconciledAbort bool `json:"reconciledAbort,omitempty"`
```

Comment:
Instead of `ReconciledAbort bool`, can we have `AbortedAt *metav1.Time` which gets cleared out whenever Status.Abort = true